### PR TITLE
Migrating ORT Extensions from Git submodule to cmake FetchContent

### PR DESCRIPTION
**Pull request by jchen351**
### Description

Merging extensions from Git submodule to cmake FetchContent

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/post-merge-jobs.yml [hunk 1]
```diff
@@ -1,3 +1,95 @@
+jobs:
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/post-merge-jobs.yml:1]: Please put the jobs into a stage.

File: tools/ci_build/github/azure-pipelines/post-merge-jobs.yml [hunk 2]
```diff
@@ -1,6 +1,100 @@
...
+
+  - template: templates/c-api-linux-cpu.yml
+    parameters:
+      AdditionalBuildFlags: --use_extensions
+      BaseImage: 'arm64v8/centos:7'
+      OnnxruntimeArch: 'aarch64'
+      OnnxruntimeCFlags: '-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -fstack-clash-protection -O3 -Wl,--strip-all'
+      OnnxruntimeCXXFlags: '-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -fstack-clash-protection -O3 -Wl,--strip-all'
+      OnnxruntimeNodejsBindingArch: 'arm64'
+      PoolName: 'aiinfra-linux-ARM64-CPU-2019'
```
Comment (mszhanyi) [inline tools/ci_build/github/azure-pipelines/post-merge-jobs.yml:23]: blank line

Comment (mszhanyi) [reply tools/ci_build/github/azure-pipelines/post-merge-jobs.yml:23]: could you please check why these post merge jobs workflows failed

File: cmake/external/extensions.cmake [hunk 3]
```diff
@@ -35,11 +35,22 @@ if (onnxruntime_WEBASSEMBLY_DEFAULT_EXTENSION_FLAGS)
   set(OCOS_ENABLE_VISION OFF CACHE INTERNAL "")
 endif()
 
-# when onnxruntime-extensions is not a subdirectory of onnxruntime,
-# output binary directory must be explicitly specified.
-# and the output binary path is the same as CMake FetchContent pattern
-add_subdirectory(${onnxruntime_EXTENSIONS_PATH} ${CMAKE_BINARY_DIR}/_deps/extensions-subbuild EXCLUDE_FROM_ALL)
+# onnxruntime-extensions
+if (NOT onnxruntime_EXTENSIONS_OVERRIDDEN)
```
Comment (wenbingl) [inline cmake/external/extensions.cmake:39]: is this flag necessary? can we just check whether onnxruntime_EXTENSIONS_PATH is defined here?

**Review by wenbingl — COMMENTED** [review]
Since we are enabling FetchContent here, it may be a good time to change overridden_path to 'overridden git commit id'?

File: cmake/external/extensions.cmake (hunk 3 above)
Comment (jchen351) [reply cmake/external/extensions.cmake:39]: The path [is always defined](https://github.com/microsoft/onnxruntime/blob/6480f665fe4875cf0238bde9dab64f0497f62200/tools/ci_build/build.py#L1233) to point to the path of extensions

File: tools/ci_build/github/azure-pipelines/post-merge-jobs.yml (hunk 2 above)
Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/post-merge-jobs.yml:23]: Someone uploaded the artifact without onnxruntime-extensions/archive/81e7799c69044c745239202085eb0a98f102937b.zip

Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/post-merge-jobs.yml:23]: Lets get this PR merged first and than create a ticket for this new feature.

File: cmake/external/extensions.cmake (hunk 3 above)
Comment (jchen351) [reply cmake/external/extensions.cmake:39]: Lets get this PR merged first and than create a ticket for this new feature.

**Review by mszhanyi — APPROVED** [review]

**Pull request merged**